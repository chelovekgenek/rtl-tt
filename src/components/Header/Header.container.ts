import { connect } from "react-redux"
import { TAppState } from "store/entities"

import { getInfo } from "store/entities/bundle"

import { Header } from "./Header"

export interface IStateProps extends ReturnType<typeof getInfo> {}

export const HeaderContainer = connect<IStateProps, {}, {}, TAppState>(state => getInfo(state))(Header)
