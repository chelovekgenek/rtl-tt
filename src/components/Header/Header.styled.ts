import styled from "styled-components"

export const Container = styled.header`
  text-align: center;
  margin-bottom: 6rem;
`

export const Image = styled.img`
  width: 100%;
  height: auto;
`
const marginMixin = `
  margin: 0 0 1rem;
`

export const Label = styled.h2`
  text-transform: uppercase;
  color: ${props => props.theme.colors.primary};
  ${marginMixin}
`
export const Title = styled.h1`
  margin: 0 0 1rem;
`
