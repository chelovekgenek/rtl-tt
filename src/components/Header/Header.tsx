import React from "react"
import { IStateProps } from "./Header.container"
import { Spinner } from "components/commons"

import * as Styles from "./Header.styled"

interface IProps extends IStateProps {}

export const Header: React.FC<IProps> = ({ fetching, data }) => {
  if (fetching) {
    return <Spinner />
  }
  if (!data) {
    return null
  }
  return (
    <Styles.Container>
      <Styles.Image src={data.image.imageUrl} alt="image" />
      <Styles.Label>{data.label}</Styles.Label>
      <Styles.Title>{data.title}</Styles.Title>
      <p>{data.description}</p>
    </Styles.Container>
  )
}
