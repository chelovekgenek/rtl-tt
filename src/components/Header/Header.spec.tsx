import React from "react"
import { shallow } from "enzyme"

import bundleInfo from "mocks/bundle-info.json"

import { Header } from "./Header"

describe("Header", () => {
  const render = (props: Partial<React.ComponentProps<typeof Header>> = {}) =>
    shallow(<Header data={bundleInfo} fetching={false} {...props} />)

  it("should render", () => {
    expect(render()).toMatchSnapshot()
  })
  it("should render Spinner while fetching is true", () => {
    expect(render({ fetching: true })).toMatchSnapshot()
  })
  it("shouldn't render anything while data is not provided", () => {
    expect(render({ fetching: false, data: undefined })).toMatchSnapshot()
  })
})
