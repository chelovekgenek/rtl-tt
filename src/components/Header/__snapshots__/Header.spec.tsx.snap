// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<styled.header>
  <styled.img
    alt="image"
    src="https://www.rtlboulevard.nl/sites/default/files/content/images/2018/12/27/thumbnail_Jaaroverzicht%20Banner.jpg"
  />
  <styled.h2>
    best gelezen
  </styled.h2>
  <styled.h1>
    De uitschieters van 2018
  </styled.h1>
  <p>
    Waar hadden we het dit jaar allemaal over? Je leest het in dit overzicht met de best gelezen artikelen.
  </p>
</styled.header>
`;

exports[`Header should render Spinner while fetching is true 1`] = `<Spinner />`;

exports[`Header shouldn't render anything while data is not provided 1`] = `""`;
